language/move-core/types/src/language_storage.rs

```diff
@@ -142,11 +192,23 @@ impl ModuleId {
 
 impl Display for ModuleId {
     fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
-        write!(f, "{}::{}", self.address, self.name)
+        self.short_str_lossless().fmt(f)
```

Comment:
We should add `0x` prefix to ModuleId str, but should not use `short_str` in the output format, because `0x1` != `0x00000000000000000000000000000001` in str presentation, this will cause some compatibility issue in offchain usage(web3 js, SDK etc).

So I suggest:

1. When input, we can accept both short and full formats.
2. When output, we just output the full format.

Reply:
I've fixed the general input of `AccountAddress` accepts `000...01`, `0x1` and `0x00001` for serialization from JSON, and from string previously.

As for the output, it's a mess and comes out in either `000000...01` or `0x1`.  We should probably align similarly to `0x000...01`

Reply:
Ok-- i'll update this. Thanks for the feedback!